### PR TITLE
feat: add test cases for POST /requests creating OOO request

### PR DESCRIPTION
**Pull request by surajmaity1**
Date: 08 Mar, 2025

Developer Name: Suraj Maity ( @surajmaity1  )

---

## Issue Ticket Number

- Open #2242 
## Description
This PR adds test cases for the POST /requests API for OOO requests to ensure its correctness and reliability. The tests cover the following scenarios:

- Valid input for creating OOO request.
- Invalid input handling (e.g., missing or incorrect fields).
- Authentication and authorization checks.
- Error handling for edge cases and unexpected failures.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

unit validator

unit service

image

</details>

## Additional Notes

- This test PR has to be merged into the Feature PR ( https://github.com/Real-Dev-Squad/website-backend/pull/2383 ), and then the feature PR ( https://github.com/Real-Dev-Squad/website-backend/pull/2383 ) can be merged into develop.

### CODE REVIEW

File: test/fixtures/oooRequest/oooRequest.ts [hunk 1]
```diff
@@ -57,8 +56,8 @@ export const createOooRequests = {
   type: "OOO",
   from: Date.now() + 100000,
   until: Date.now() + 200000,
-  message: "Out of office for personal reasons.",
-  state: REQUEST_STATE.PENDING,
+  reason: "Out of office for personal reasons.",
+  status: REQUEST_STATE.PENDING,
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/fixtures/oooRequest/oooRequest.ts:60]: This OOO request has been used in acknowledging the OOO request. It will be fixed in acknowledging the OOO request PR.

File: test/fixtures/oooRequest/oooRequest.ts [hunk 2]
```diff
@@ -15,8 +15,7 @@
   type: "OOO",
   from: Date.now() + 100000,
   until: Date.now() + 200000,
-  message: "Out of office for personal reasons.",
-  state: REQUEST_STATE.PENDING,
+  reason: "Out of office for personal reasons.",
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/fixtures/oooRequest/oooRequest.ts:18]: This OOO request has been used in acknowledging the OOO request. It will be fixed in acknowledging the OOO request PR.

File: test/unit/services/oooRequest.test.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
+import sinon from "sinon";
+import cleanDb from "../../utils/cleanDb";
+import {
+    OOO_STATUS_ALREADY_EXIST,
+    REQUEST_ALREADY_PENDING,
+    USER_STATUS_NOT_FOUND,
+} from "../../../constants/requests";
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts:7]: In this Feature PR https://github.com/Real-Dev-Squad/website-backend/pull/2383 , I included this.

File: test/unit/services/oooRequest.test.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+import { convertDaysToMilliseconds } from "../../../utils/time";
+import { createOOORequest, validateUserStatus } from "../../../services/oooRequest";
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts:9]: In this Feature PR https://github.com/Real-Dev-Squad/website-backend/pull/2383 , I included this.

**Review by AnujChhikara — COMMENTED** [review]
Can you please add test coverage report?

**Review by AnujChhikara — COMMENTED** [review]
Hello @surajmaity1, since you've written tests for validators, middleware, and controllers, could you please include the coverage for each section separately? The screenshot you attached doesn’t specify which section the test coverage belongs to. Thanks!

File: test/integration/requests.test.ts
Comment (AnujChhikara) [inline test/integration/requests.test.ts:1]: Can you please write a test to fetch OOO requests using the state query parameter and verify the fields returned in the response?

File: test/integration/requests.test.ts [hunk 5]
```diff
@@ -61,92 +69,115 @@ describe("/requests OOO", function () {
...
+
+          requestsQuery.getRequestByKeyValues({
+            userId: testUserId,
+            type: REQUEST_TYPE.OOO,
+            status: REQUEST_STATE.PENDING
+          }).then((request) => {
+            expect(request).to.not.be.null;
+            expect(request.reason).to.equal(validOooStatusRequests.reason);
+            done();
+          }).catch(done);
```
Comment (pankajjs) [inline test/integration/requests.test.ts:172]: Why are we not using async await syntax here?

File: test/integration/requests.test.ts [hunk 6]
```diff
@@ -157,30 +188,32 @@ describe("/requests OOO", function () {
         });
     });
 
-    it("should return error if message is not present in body", function (done) {
+    it("should return 400 with 'reason is required' message when reason is missing", function (done) {
```
Comment (pankajjs) [inline test/integration/requests.test.ts:191]:
```suggestion
    it("should return 400 when reason field is missing in request body", function (done) {
```

File: test/integration/requests.test.ts [hunk 7]
```diff
@@ -157,30 +188,32 @@ describe("/requests OOO", function () {
...
           expect(res).to.have.status(400);
           expect(res.body).to.have.property("message");
-          expect(res.body.message).to.equal("message is required");
+          expect(res.body.message).to.equal("reason is required");
           done();
         });
     });
 
-    it("should return error if state in the body is not PENDING", function (done) {
+    it("should return 400 with error when status field is included in request body", function (done) {
```
Comment (pankajjs) [inline test/integration/requests.test.ts:206]: Can you please write test to validate other fields such as from and until?

File: test/integration/requests.test.ts [hunk 8]
```diff
@@ -157,30 +188,32 @@ describe("/requests OOO", function () {
...
+        .post(`${requestsEndpoint}?dev=true`)
         .set("cookie", `${cookieName}=${authToken}`)
-        .send({ ...validOooStatusRequests, state: REQUEST_STATE.APPROVED })
+        .send({ ...validOooStatusRequests, status: REQUEST_STATE.APPROVED })
         .end(function (err, res) {
+          if (err) return done(err);
           expect(res).to.have.status(400);
           expect(res.body).to.have.property("message");
-          expect(res.body.message).to.equal("state must be PENDING");
+          expect(res.body.message).to.equal(`"status" is not allowed`);
```
Comment (pankajjs) [inline test/integration/requests.test.ts:216]: Can you please write test for these cases?
1. When user does not have user status.
2. When user is already ooo.
3. When user has a pending request

File: test/unit/services/oooRequest.test.ts [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+    });
+
+    describe("validateUserStatus", function() {
+
+        it("should return USER_STATUS_NOT_FOUND if user status not found", async function() {
+            try {
+                await validateUserStatus(
+                    testUserId,
+                    false,
+                    null
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:45]: Why do we have 3 arguments?

File: test/unit/services/oooRequest.test.ts [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+            testUserStatus = {
+                userId: testUserId,
+                currentStatus: {
+                    updatedAt: Date.now() - convertDaysToMilliseconds(5),
+                    from: Date.now(),
+                    until: Date.now() + convertDaysToMilliseconds(5),
+                    message: "taking OOO for personal reason",
+                    state: "OOO",
+                }
+            };
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:64]: Can we create this in fixture?

File: test/unit/services/oooRequest.test.ts [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+            userId: testUserId,
+            currentStatus: {
+                updatedAt: Date.now(),
+                from: Date.now(),
+                until: "",
+                message: "",
+                state: "ACTIVE",
+            }
+        };
+    });
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:31]: Can we create this in fixture?

File: test/unit/services/oooRequest.test.ts [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+                expect(error.message).to.equal(OOO_STATUS_ALREADY_EXIST);
+            }
+        });
+
+        it("should return undefined when all validation checks passes", async function() {
+            const response = await validateUserStatus(
+                testUserId,
+                true,
+                testUserStatus
+            );
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:83]: Why do we have 3 arguments?

File: test/unit/services/oooRequest.test.ts [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+        it("should throw error", async function() {
+            sinon.stub(logService, "addLog").throws(new Error(errorMessage));
+            const validateSpy = sinon.spy(require("../../../services/oooRequest"), "validateUserStatus");
+
+            try {
+                await validateUserStatus(
+                    testUserId,
+                    true,
+                    testUserStatus
+                );
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:96]: RE: Why do we have 3 arguments?

File: test/unit/services/oooRequest.test.ts [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+                testUserId,
+                true,
+                testUserStatus
+            );
+            expect(response).to.not.exist;
+        });
+
+        it("should throw error", async function() {
+            sinon.stub(logService, "addLog").throws(new Error(errorMessage));
+            const validateSpy = sinon.spy(require("../../../services/oooRequest"), "validateUserStatus");
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:89]: Why this?

File: test/unit/services/oooRequest.test.ts [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+                requestedBy: "test-username-1",
+                userId: testUserId,
+                comment: null
+            });
+        });
+
+        it("should throw error", async function() {
+            sinon.stub(logService, "addLog").throws(new Error(errorMessage));
+            const createSpy = sinon.spy(require("../../../services/oooRequest"), "createOOORequest");
+
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:148]: RE: why this?

File: test/unit/services/oooRequest.test.ts [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+                    testUserStatus
+                );
+            } catch(error) {
+                expect(error.message).to.equal(errorMessage);
+                expect(validateSpy.calledOnce).to.be.true;
+            }
+        });
+    });
+
+    describe("createOOORequest", function() {
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:104]: Can you please write tests for these cases?
 - when user status is not found
 - when user status is already oooo

File: test/unit/services/oooRequest.test.ts (hunk 15 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts:148]: In this test case, createSpy is used to verify that the `createOOORequest ` function (imported from `../../../services/oooRequest`) was actually called during the test's execution.

This allows us to track if and how many times createOOORequest is called.

File: test/unit/services/oooRequest.test.ts (hunk 16 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts:104]: these test cases are written here - https://github.com/Real-Dev-Squad/website-backend/blob/bf14b5fbc9cc04dfaf656c386fc5e653ddca4e04/test/unit/services/oooRequest.test.ts#L40

File: test/unit/services/oooRequest.test.ts (hunk 14 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts:89]: replied here - https://github.com/Real-Dev-Squad/website-backend/pull/2386#discussion_r2013216659

File: test/integration/requests.test.ts [hunk 17]
```diff
@@ -41,12 +44,17 @@ let approvedOooRequestId: string;
 let oooRequestData: any;
 let oooRequestData2: any;
 let testUserId: string;
+let testUserIdForNotDiscordUser: string;
 
 describe("/requests OOO", function () {
+
+  const requestsEndpoint: string = "/requests";
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/integration/requests.test.ts:51]: Previously, I wrote like this. After discussing with @AnujChhikara , he told that we can use "/requests", which will be reused for GET requests as well.

File: test/integration/requests.test.ts [hunk 18]
```diff
@@ -61,92 +69,115 @@ describe("/requests OOO", function () {
...
-      await chai
-        .request(app)
-        .post("/requests")
-        .set("cookie", `${cookieName}=${authToken}`)
-        .send(validOooStatusRequests);
-      const response = await chai
+    it("should return 401 if user is not part of discord", function (done) {
+      const authTokenForNotDiscordUser = authService.generateAuthToken(
+        { userId: testUserIdForNotDiscordUser }
+      );
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/integration/requests.test.ts:121]: resolved.

File: test/integration/requests.test.ts [hunk 19]
```diff
@@ -41,12 +44,17 @@ let approvedOooRequestId: string;
...
 
 describe("/requests OOO", function () {
+
+  const requestsEndpoint: string = "/requests";
+
   beforeEach(async function () {
-    const userIdPromises = [addUser(userData[16]), addUser(userData[4])];
-    const [userId, superUserId] = await Promise.all(userIdPromises);
+    const userIdPromises = [addUser(userData[16]), addUser(userData[4]), addUser(userData[18])];
+    const [userId, superUserId, userIdForNotDiscordUser] = await Promise.all(userIdPromises);
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/integration/requests.test.ts:55]: Fixed

File: test/integration/requests.test.ts (hunk 5 above)
Comment (surajmaity1) [reply test/integration/requests.test.ts]: Fixed.

File: test/unit/services/oooRequest.test.ts (hunk 13 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Hi @pankajjs , I have fixed this.

File: test/unit/services/oooRequest.test.ts (hunk 12 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Hi @pankajjs , I have fixed this.

File: test/integration/requests.test.ts [hunk 20]
```diff
@@ -61,92 +69,115 @@ describe("/requests OOO", function () {
...
         .request(app)
-        .post("/requests")
+        .post(`${requestsEndpoint}?dev=true`)
         .set("cookie", `${cookieName}=${authToken}`)
         .send(validOooStatusRequests)
         .end(function (err, res) {
+          if (err) return done(err);
           expect(res).to.have.status(201);
           expect(res.body).to.have.property("message");
+          expect(Object.keys(res.body)).to.have.lengthOf(1);
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/integration/requests.test.ts]: it's removed.

File: test/integration/requests.test.ts [hunk 21]
```diff
@@ -61,92 +69,115 @@ describe("/requests OOO", function () {
...
-      expect(response).to.have.status(400);
-      expect(response.body).to.have.property("message");
-      expect(response.body.message).to.equal(REQUEST_ALREADY_PENDING);
+        .post(`${requestsEndpoint}?dev=true`)
+        .set("cookie", `${cookieName}=${authTokenForNotDiscordUser}`)
+        .send(validOooStatusRequests)
+        .end(function (err, res) {
+          expect(res).to.have.status(401);
+          expect(res.body.error).to.equal("Unauthorized");
+          expect(res.body.message).to.equal("Only discord user can create an OOO request");
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/integration/requests.test.ts]: yes, it's fixed.
Thanks

File: test/integration/requests.test.ts [hunk 22]
```diff
@@ -61,92 +69,115 @@ describe("/requests OOO", function () {
...
     });
 
-    it("should return 400, if already created request is created again", async function () {
-      await chai
-        .request(app)
-        .post("/requests")
-        .set("cookie", `${cookieName}=${authToken}`)
-        .send(validOooStatusRequests);
-      const response = await chai
+    it("should return 401 if user is not part of discord", function (done) {
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/integration/requests.test.ts]: Hi @pankajjs , it is fixed.

File: test/integration/requests.test.ts (hunk 6 above)
Comment (surajmaity1) [reply test/integration/requests.test.ts]: Fixed

File: test/integration/requests.test.ts (hunk 7 above)
Comment (surajmaity1) [reply test/integration/requests.test.ts:255]: Hi @pankajjs , it's added

File: test/integration/requests.test.ts (hunk 8 above)
Comment (surajmaity1) [reply test/integration/requests.test.ts:265]: Hi @pankajjs , I wrote all test cases. Thanks

File: test/unit/services/oooRequest.test.ts (hunk 10 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Fixed.

File: test/unit/services/oooRequest.test.ts (hunk 11 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Fixed.

File: test/unit/services/oooRequest.test.ts (hunk 9 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Hi @pankajjs , it's fixed. Thanks

File: test/integration/requests.test.ts
Comment (surajmaity1) [reply test/integration/requests.test.ts:1]: It will not work using a state query. We have to change the `state` field to the `status `field. 

GitHub issue: https://github.com/Real-Dev-Squad/website-backend/issues/2399

File: test/unit/services/oooRequest.test.ts [hunk 23]
```diff
@@ -0,0 +1,135 @@
...
+                reason: validOooStatusRequests.reason,
+                status: "PENDING",
+                lastModifiedBy: null,
+                requestedBy: "test-username-1",
+                userId: testUserId,
+                comment: null
+            });
+        });
+    });
+});
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:135]: The `createOOORequest` depends on multiple functions such as `getUserStatus `, `validateUserStatus`, `getRequestByKeyValues ` and `createRequest `. Please make sure to correctly mock all dependencies while testing the function.

File: test/unit/services/oooRequest.test.ts [hunk 24]
```diff
@@ -0,0 +1,135 @@
...
+import { expect } from "chai";
+import { validOooStatusRequests, validUserCurrentStatus } from "../../fixtures/oooRequest/oooRequest";
+import { deleteUserStatus, updateUserStatus } from "../../../models/userStatus";
+import { userState } from "../../../constants/userStatus";
+
+describe("Test OOO Request Service", function() {
+
+    const testUserId = "11111";
+    const testUserStatusId = "22222";
+    let testUserStatus;
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:18]: type missing

**Review by pankajjs — CHANGES_REQUESTED** [review]

File: test/unit/services/oooRequest.test.ts (hunk 24 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: fixed.

File: test/unit/services/oooRequest.test.ts [hunk 25]
```diff
@@ -0,0 +1,149 @@
...
+                expect(error.statusCode).to.equal(409);
+                expect(error.message).to.equal(REQUEST_ALREADY_PENDING);
+            }
+        });
+
+        it("should create OOO request", async function() {
+            const response = await createOOORequest(
+                validOooStatusRequests,
+                testUserName,
+                testUserId
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:134]: dependencies are not mocked correctly

File: test/unit/services/oooRequest.test.ts [hunk 26]
```diff
@@ -0,0 +1,149 @@
...
+        it("should return REQUEST_ALREADY_PENDING if user has already pending request", async function() {
+            const mockResponse = {
+                id: 'qcl0ZLsnngKUNZY9GkCo',
+                userId: testUserId,
+                type: REQUEST_TYPE.OOO,
+                status: REQUEST_STATE.PENDING,
+            };
+
+            const getRequestByKeyValuesStub = sinon.stub(requestModel, "getRequestByKeyValues").resolves(mockResponse);
+
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:120]: where are other mock dependencies?

File: test/unit/services/oooRequest.test.ts [hunk 27]
```diff
@@ -0,0 +1,149 @@
...
+        it("should return OOO_STATUS_ALREADY_EXIST if user status is already OOO", async function() {
+            const getUserStatusStub = sinon.stub(userStatus, 'getUserStatus').resolves({
+                userStatusExists: true,
+                data: {
+                    currentStatus: validUserCurrentStatus,
+                },
+            });
+            try {
+                await createOOORequest(validOooStatusRequests, testUserName, testUserId);
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:103]: Where are other mock dependencies?

File: test/unit/services/oooRequest.test.ts [hunk 28]
```diff
@@ -0,0 +1,149 @@
...
+
+        afterEach(async function () {
+            sinon.restore();
+        });
+
+        it("should return USER_STATUS_NOT_FOUND if user status not found", async function() {
+            const getUserStatusStub = sinon.stub(userStatus, 'getUserStatus').resolves({ userStatusExists: false });
+            try {
+                await createOOORequest(validOooStatusRequests, testUserName, testUserId);
+            } catch (error) {
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:88]: Where are other mock dependencies?

File: test/unit/services/oooRequest.test.ts [hunk 29]
```diff
@@ -0,0 +1,149 @@
...
+const requestModel = require("../../../models/requests");
+
+describe("Test OOO Request Service", function() {
+
+    let testUserName: string;
+    let testUserId: string;
+
+    beforeEach(async function() {
+        const user = userDataFixture()[8];
+        const userId = await addUser(user);
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:27]: Do we need these variable? These are not being used any where apart from this block.

File: test/unit/services/oooRequest.test.ts [hunk 30]
```diff
@@ -0,0 +1,149 @@
...
+const userStatus = require("../../../models/userStatus");
+const requestModel = require("../../../models/requests");
+
+describe("Test OOO Request Service", function() {
+
+    let testUserName: string;
+    let testUserId: string;
+
+    beforeEach(async function() {
+        const user = userDataFixture()[8];
```
Comment (pankajjs) [inline test/unit/services/oooRequest.test.ts:26]: Refactor this line.

**Review by pankajjs — CHANGES_REQUESTED** [review]

File: test/unit/services/oooRequest.test.ts (hunk 29 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts:27]: We're taking userId and username that will be used in the validateUserStatus & createOOORequest functions.

File: test/unit/services/oooRequest.test.ts (hunk 30 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Hi @pankajjs , it's fixed now.
Thank you

File: test/unit/services/oooRequest.test.ts (hunk 29 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Hi @pankajjs , it's fixed now.
Thank you

File: test/unit/services/oooRequest.test.ts (hunk 28 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Hi @pankajjs , it's fixed now.
Thank you

File: test/unit/services/oooRequest.test.ts (hunk 27 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Hi @pankajjs , it's fixed now.
Thank you

File: test/unit/services/oooRequest.test.ts (hunk 25 above)
Comment (surajmaity1) [reply test/unit/services/oooRequest.test.ts]: Hi @pankajjs , it's fixed now.
Thank you

**Comment by prakashchoudhary07** [comment]
@surajmaity1 Failing test case?

**Comment by surajmaity1** [comment]
> @surajmaity1 Failing test case?

Hi @prakashchoudhary07 , test cases are failing becuase this PR only contains tests of feature PR ( https://github.com/Real-Dev-Squad/website-backend/pull/2383 ) .

**Comment by prakashchoudhary07** [comment]
> > @surajmaity1 Failing test case?
> 
> Hi @prakashchoudhary07 , test cases are failing becuase this PR only contains tests of feature PR ( #2383 ) .

Then shouldn't it be mentioned in the description?

Also, what is the plan? When are we going to merge this PR
And special note please mention it in the description

**Comment by surajmaity1** [comment]
> > > @surajmaity1 Failing test case?
> > 
> > 
> > Hi @prakashchoudhary07 , test cases are failing becuase this PR only contains tests of feature PR ( #2383 ) .
> 
> Then shouldn't it be mentioned in the description?
> 
> Also, what is the plan? When are we going to merge this PR And special note please mention it in the description

Hi @prakashchoudhary07 , can you please check now in the additional notes section?

File: test/fixtures/oooRequest/oooRequest.ts [hunk 31]
```diff
@@ -19,6 +20,35 @@ export const validOooStatusRequests = {
...
+//   type: validOooStatusRequests.type,
+//   from: validOooStatusRequests.from,
+//   until: validOooStatusRequests.until,
+//   reason: validOooStatusRequests.reason,
+//   status: "PENDING",
+//   lastModifiedBy: null,
+//   requestedBy: "suraj-maity-1",
+//   userId: "jCqqOYCnm93mcmaYuSsQ",
+//   comment: null
+// };
```
Comment (shobhan-sundar-goutam) [inline test/fixtures/oooRequest/oooRequest.ts:34]: Why files are committed with these comments?

File: test/fixtures/oooRequest/oooRequest.ts [hunk 32]
```diff
@@ -19,6 +20,35 @@ export const validOooStatusRequests = {
...
+  updatedAt: Date.now(),
+};
+
+// export const testUserStatus: UserStatus = {
+//   id: "wcl0ZLsnngKUNZY9GkCo",
+//   data: {
+//       currentStatus: validUserCurrentStatus
+//   },
+//   userStatusExists: true
+// };
```
Comment (shobhan-sundar-goutam) [inline test/fixtures/oooRequest/oooRequest.ts:50]: Same here. Why are we keeping the comments?

Comment (surajmaity1) [reply test/fixtures/oooRequest/oooRequest.ts:50]: At line 44, the `UserStatus` is currently unavailable in this branch. Attempting to uncomment this line will result in a compilation error: "Cannot find name 'UserStatus'", as it is defined in the feature pull request.

File: test/fixtures/oooRequest/oooRequest.ts (hunk 31 above)
Comment (surajmaity1) [reply test/fixtures/oooRequest/oooRequest.ts:34]: At line 28, the `reason` identifier is currently unavailable in this branch. Attempting to uncomment this line will result in a compilation error: "Cannot find name 'reason'", as it is defined in the feature pull request.

File: test/integration/requests.test.ts [hunk 33]
```diff
@@ -69,137 +81,239 @@ describe("/requests OOO", function () {
...
+      await deleteUserStatus(testUserId);
+      const response = await chai
+        .request(app)
+        .post(requestsEndpoint)
+        .set("cookie", `${cookieName}=${authToken}`)
+        .send(validOooStatusRequests);
+
+      expect(response).to.have.status(404);
+      expect(response.body).to.have.property("message");
+      // expect(response.body.message).to.equal(USER_STATUS_NOT_FOUND);
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/integration/requests.test.ts:280]: At line 280, the `USER_STATUS_NOT_FOUND` is currently unavailable in this branch. Attempting to uncomment this line will result in a compilation error: "Cannot find name 'USER_STATUS_NOT_FOUND'", as it is defined in the feature pull request.

File: test/integration/requests.test.ts [hunk 34]
```diff
@@ -69,137 +81,239 @@ describe("/requests OOO", function () {
...
+      await updateUserStatus(testUserId, testOOOUserStatus);
+      const response = await chai
+        .request(app)
+        .post(requestsEndpoint)
+        .set("cookie", `${cookieName}=${authToken}`)
+        .send(validOooStatusRequests);
+
+      expect(response).to.have.status(403);
+      expect(response.body).to.have.property("message");
+      // expect(response.body.message).to.equal(OOO_STATUS_ALREADY_EXIST);
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/integration/requests.test.ts:298]: At line 298, the `OOO_STATUS_ALREADY_EXIST`  is currently unavailable in this branch. Attempting to uncomment this line will result in a compilation error: "Cannot find name 'OOO_STATUS_ALREADY_EXIST'", as it is defined in the feature pull request.

File: test/integration/requests.test.ts [hunk 35]
```diff
@@ -69,137 +81,239 @@ describe("/requests OOO", function () {
...
+        .post(requestsEndpoint)
+        .set("cookie", `${cookieName}=${authToken}`)
+        .send(validOooStatusRequests);
+      const response = await chai
+        .request(app)
+        .post(requestsEndpoint)
+        .set("cookie", `${cookieName}=${authToken}`)
+        .send(validOooStatusRequests);
+
+      expect(response).to.have.status(400);
```
[the remark this answers is not in the document]
Comment (surajmaity1) [reply test/integration/requests.test.ts]: Fixed.

**Review by AnujChhikara — APPROVED** [review]
All my comments have been addressed. Since this PR is not stacked on the feature PR, we’ve had to temporarily skip/comment out some tests because the corresponding feature code isn't yet present in the repo.
I'll personally ensure that these test cases are uncommented once the feature PR is under review or merged.

Approving this PR

**Review by prakashchoudhary07 — APPROVED** [review]
LGTM
